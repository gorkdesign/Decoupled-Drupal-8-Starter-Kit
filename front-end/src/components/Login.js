import React from 'react';
import PropTypes from 'prop-types';

import '../styles/signin.css';

const Login = props => (
  <div className="signin-container">
    <div className="text-center">
      <form className="form-signin" onSubmit={props.handleLogin}>
        <h1 className="h3 mb-3 font-weight-normal">Please sign in</h1>
        <input
          className="form-control username"
          name="username"
          type="username"
          placeholder="Username"
          onChange={props.handleInputChange}
        />
        <input
          className="form-control password"
          name="password"
          type="password"
          placeholder="Password"
          onChange={props.handleInputChange}
        />
        <div className="checkbox mb-3">
          <input type="checkbox" value="remember-me" /> Remember me
        </div>
        <button className="btn btn-lg btn-primary btn-block submit" type="submit">Sign in</button>
      </form>
      { props.isLoginFailed ? <p className="error" style={{ color: 'red', display: props.isLoginFailed ? 'block' : 'none' }}>Credentials incorrect</p> : null }
    </div>
  </div>
);

Login.propTypes = {
  handleInputChange: PropTypes.func.isRequired,
  handleLogin: PropTypes.func.isRequired,
  isLoginFailed: PropTypes.bool.isRequired,
};

export default Login;
