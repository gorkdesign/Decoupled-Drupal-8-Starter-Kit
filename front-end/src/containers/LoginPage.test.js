import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Redirect } from 'react-router-dom';

import Login from './LoginArticle';


describe('LoginArticle', () => {
  const initialPropsState = {
    username: '',
    password: '',
    dispatch: jest.fn(),
    isLoggingIn: false,
    isAuthenticated: false,
    error: '',
  };

  let props;
  let mountedLogin;
  const loginArticle = (modifiedProps = initialPropsState) => {
    props = {
      ...initialPropsState,
      ...modifiedProps,
    };

    if (!mountedLogin) {
      mountedLogin = mount(<Login {...props} />);
    }
    return mountedLogin;
  };

  beforeEach(() => {
    props = initialPropsState;
    mountedLogin = undefined;
  });

  describe('Snapshots', () => {
    const output = shallow(<Login {...initialPropsState} />);
    it('should render correctly', () => {
      expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('state should be updated when username changes', () => {
      const component = loginArticle();
      component.find('.username').simulate('change', { target: { name: 'username', value: 'test' } });
      expect(component.state('username') === 'test');
    });

    it('state should be updated when password changes', () => {
      const component = loginArticle();
      component.find('.password').simulate('change', { target: { name: 'password', value: 'test' } });
      expect(component.state('password') === 'test');
    });

    it('should respond to submit button press', () => {
      const component = loginArticle();
      component.find('[type="submit"]').simulate('submit');
      expect(props.dispatch.mock.calls.length === 1);
    });

    it('submit should be disabled until username and password are entered', () => {
      const component = loginArticle();
      component.find('.username').simulate('change', { target: { name: 'username', value: 'test' } });
      component.find('[type="submit"]').simulate('submit');
      expect(component.state('error').length > 0).toBe(true);
    });
  });


  describe('LoginArticle differing props', () => {
    it('Should NOT display the login form if authenticated - SHOULD REDIRECT', () => {
      const newProps = { ...initialPropsState };
      newProps.isAuthenticated = true;
      const wrapper = shallow(<Login {...newProps} />);
      expect(wrapper.find(Redirect).props().to).toBe('/');
    });

    it('Do NOT displays an error message when `error` is NOT passed in', () => {
      const component = loginArticle({ isLoggingIn: true });
      expect(component.find('.loggingIn').length).toBe(1);
    });
  });
});
