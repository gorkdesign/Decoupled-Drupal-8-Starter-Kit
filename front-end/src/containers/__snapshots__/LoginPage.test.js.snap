// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Container Snapshots should render correctly 1`] = `
<Login
  dispatch={[Function]}
  error=""
  handleInputChange={[Function]}
  handleLogin={[Function]}
  isAuthenticated={false}
  isLoggingIn={false}
  password=""
  username=""
/>
`;
